src/pet.js

```diff
@@ -0,0 +1,89 @@
+const MAX_FITNESS = 10
+const MIN_HUNGER = 0
+
+const GROWUP_HUNGER_FACTOR = 5
+const GROWUP_FITNESS_FACTOR = 3
+const GROWUP_AGE_FACTOR = 1
+const WALK_FITNESS_FACTOR = 4
+const FEED_HUNGER_INCREMENT = 3
+
+class Pet {
+    constructor(name) {
+        this.name = name; 
+        this.age = 0;
+        this.hunger = 0;
+        this.fitness = 10;
+        this.children = [];
+    }
+    get isAlive() {
+        return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    } 
+}
+   
+    Pet.prototype.adoptChild = function (child) {
+        if (!this.isAlive) {
+            throw new Error('Your child pet is no longer alive!!');
+        }
+        this.children.push(child);
+    }
+    Pet.prototype.haveBaby = function (baby) {
```

Comment:
You should add a clear line between each of the methods here to break up the solid lump of code you currently have. Just makes it easier to read
```
    }

    Pet.prototype.haveBaby = function (baby) {
```